### PR TITLE
CI: Making sure backend tests run when expected to run

### PR DESCRIPTION
**Pull request by datapythonista**
Some backend tests are currently not running, and it's difficult to be sure they are with the current code. This PR introduces some improvements to make sure tests are run when asked to run, and they are not skipped silently:
- Fixes typo when creating list of backend tests to run
- Raises an error if an unknown backend is requested to run in `PYTEST_BACKENDS` (if `postgresql` is used instead of `postgres` an error will be reported instead of not running the `postgres` tests)
- Renaming the postgres test classes and markers to match the directory of the backend, as in the other backends
- Avoid skipping the tests when dependencies are not found. If the dependencies are not installed, the build will fail, instead of skipping the tests silently

### CODE REVIEW

**Comment by datapythonista** [comment]
@jreback there are some problems when calling the tests in the CI currently. I introduced an error when moving to GitHub actions, and before that, I think several things were unintentionally skipped.

This PR fixes the issues, and shouldn't let things be skipped silently as easily as before. But one of the builds has few tests failing:
```
FAILED ibis/tests/all/test_string.py::test_string[OmniSciDB-length] - Excepti...
FAILED ibis/tests/all/test_temporal.py::test_strftime[PySpark-%Y%m%d-%Y%m%d]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-01-6-Sunday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-02-0-Monday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-03-1-Tuesday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-04-2-Wednesday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-05-3-Thursday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-06-4-Friday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_scalar[PySpark-2017-01-07-5-Saturday]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_column[PySpark] - py...
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_column_group_by[PySpark-<lambda>-<lambda>0]
FAILED ibis/tests/all/test_temporal.py::test_day_of_week_column_group_by[PySpark-<lambda>-<lambda>1]
FAILED ibis/expr/tests/test_interactive.py::TestInteractiveUse::test_default_limit
```

I don't have time to continue with this PR this week, will continue next one. But to keep you aware of the unreliability of the CI at the moment. Not sure if it'd be worth to merge this PR now, even if it'll turn the CI (genuinely) red.

**Comment by jreback** [comment]
can u xfail that test and open an issue to have the CI green?

@xmnlab if you would take a look

**Comment by datapythonista** [comment]
@jreback xfailed the tests, all green here.

If I'm not wrong, @xmnlab just started holidays, and will be mostly disconnected for the next few weeks.

File: ibis/expr/tests/test_interactive.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def test_repr_png_is_not_none_in_not_interactive(self):
         ), config.option_context('graphviz_repr', True):
             assert table._repr_png_() is not None
 
+    @pytest.mark.xfail(strict=False)  # #2337
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:51]: can you add a helpful message here

File: ci/azure/linux.yml [hunk 2]
```diff
@@ -73,15 +62,17 @@ jobs:
...
-          pytest ibis -m "${PYTEST_MARK_EXPRESSION}" \
-                      -k"not compile and not connect" \
-                      -ra \
-                      --doctest-modules \
-                      --doctest-ignore-import-errors \
-                      --junitxml=/tmp/test-reports/pytest/junit.xml \
-                      --cov=ibis \
-                      --cov-report=xml:/tmp/test-reports/pytest-cov/coverage.xml
+          pytest ibis/tests ibis/expr/tests ibis/sql/tests ibis/impala/tests ibis/clickhouse/tests \
+            -m "not udf" \
```
Comment (jreback) [inline ci/azure/linux.yml:68]: shoudn't this skip int he tests themselves? (the udf)

File: ci/azure/windows.yml [hunk 3]
```diff
@@ -132,7 +133,7 @@ jobs:
 
     - script: |
         call activate $(conda.env)
-        pytest --tb=short --junitxml="junit-$(python.version).xml" -m "not backend and not clickhouse and not impala and not hdfs and not bigquery and not omniscidb and not postgis and not postgresql" -ra ibis
+        pytest ibis/tests ibis/expr/tests ibis/sql/tests ibis/pandas/tests ibis/pandas/execution/tests ibis/file/tests -m "not hdfs" -ra --junitxml="junit-$(python.version).xml"
```
Comment (jreback) [inline ci/azure/windows.yml:136]: same here shouldn't this be int he tests (skip hdfs)

File: ibis/expr/tests/test_interactive.py [hunk 4]
```diff
@@ -48,6 +48,12 @@ def test_repr_png_is_not_none_in_not_interactive(self):
         ), config.option_context('graphviz_repr', True):
             assert table._repr_png_() is not None
 
+    # XXX This test is failing in the OmniSciDB/Spark build, and working
+    # in the rest, even if does not seem to depend on the backend.
+    # For some reason in that build the statement does not contain
+    # the LIMIT. Xfailing with `strict=False` since in the other backends
+    # it does work. See #2337
+    @pytest.mark.xfail(strict=False)
```
Comment (jreback) [inline ibis/expr/tests/test_interactive.py:56]: comment is good, mean add `reason=...` in the xfail itself

File: ci/azure/linux.yml (hunk 2 above)
Comment (datapythonista) [reply ci/azure/linux.yml:68]: Not sure I understand. Do you mean that the udf tests could be skipped depending on the backend with the decorators as the rest of the tests? Probably, but it's been like this until now, and I guess some extra refactoring is needed. I can open an issue if this is what you mean. Not so sure if makes sense for hdfs.

Comment (jreback) [reply ci/azure/linux.yml:68]: yes that's exactly what i mean, e.g. the CI should simply run the backend w/o *any* logic for which marks it should run / not run (these are in the code). but yeah let's open an issue and not block this PR.

**Comment by jreback** [comment]
lgtm merge on green. (comment to create a new issue)

**Comment by datapythonista** [comment]
Thanks @jreback, created #2341

**Pull request merged**